src/sftp.rs

```diff
@@ -383,6 +383,54 @@ impl Sftp {
     pub fn fs(&self) -> Fs {
         Fs::new(self.handle.clone().write_end(), "".into())
     }
+
+    /// Check if the remote server supports the limits extension.
+    pub fn support_limits(&self) -> bool {
+        self.handle
+            .get_auxiliary()
+            .extensions()
+            .contains(Extensions::LIMITS)
+    }
+
+    /// Check if the remote server supports the expand path extension.
+    pub fn support_expand_path(&self) -> bool {
+        self.handle
+            .get_auxiliary()
+            .extensions()
+            .contains(Extensions::EXPAND_PATH)
+    }
+
+    /// Check if the remote server supports the fsync extension.
+    pub fn support_fsync(&self) -> bool {
+        self.handle
+            .get_auxiliary()
+            .extensions()
+            .contains(Extensions::FSYNC)
+    }
+
+    /// Check if the remote server supports the hardlink extension.
+    pub fn support_hardlink(&self) -> bool {
+        self.handle
+            .get_auxiliary()
+            .extensions()
+            .contains(Extensions::HARDLINK)
+    }
+
+    /// Check if the remote server supports the posix rename extension.
+    pub fn support_posix_rename(&self) -> bool {
+        self.handle
+            .get_auxiliary()
+            .extensions()
+            .contains(Extensions::POSIX_RENAME)
+    }
+
+    /// Check if the remote server supports the copy data extension.
```

Comment:
Can we also add doc to corresponded function?

E.g. for this extension:

```suggestion
    /// Check if the remote server supports the copy data extension.
    /// If it returns true, then `file::File::copy_all_to` is supported.
```

And also update doc for `file::File::copy_all_to` to point to this for checking whether it is supported?